src/base/bittorrent/torrenthandle.h

```diff
@@ -445,14 +445,26 @@ namespace BitTorrent
 
         QHash<QString, TrackerInfo> m_trackerInfos;
 
-        enum StartupState
+        enum ProcessingState
         {
-            NotStarted,
-            Starting,
-            Started
+            PreparingPaused, // torrent is preparing to start processing in "paused" state
+            PreparingResumed, // torrent is preparing to start processing in "resumed" state
+            Starting, // torrent is prepared and starting to perform regular processing
+            Working, // torrent is performing regular processing
+            CheckingPaused // torrent is checking in "paused" state
         };
+        // Handle "extra" states of torrent processing.
+        // Initially torrent can be either in "PreparingPaused" or "PreparingResumed" state.
+        // Torrent is become "paused" after checking to perform all needed preparations.
+        // Torrent that is prepared to start processing in "paused" state becomes "Working"
+        // immediatelly after the preparations are done.
+        // Torrent that is prepared to start processing in "resumed" state becomes "Starting"
+        // until it is really started.
```

Comment:
and what about the relations of `Starting` to `Working`? please comment about it.